### PR TITLE
Replace acronyms with full words

### PR DESCRIPTION
**Pull request by Tal-Dahann**
# Description

As suggested in issue #1254, added a setting to replace all ancronyms in a thread (including title, post and comments) with full words.

> For example: AITA -> Am I The Asshole

- Added `utils\acronyms.json` It's very minimal, but users are free to add their own (and its stated when configuring the setting)
- Added `replace_ancronyms` setting, default is set to False. Setting to true will replace all the acronyms with full words.
- Changed the `engine_wrapper.py` to replace the ancronyms if the setting is set to True, right after adding periods

# Issue Fixes

Not an issue but a feature

# Checklist:

- [X] I am pushing changes to the **develop** branch
- [X] I am using the recommended development environment
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have formatted and linted my code using python-black and pylint
- [X] I have cleaned up unnecessary files
- [X] My changes generate no new warnings
- [X] My changes follow the existing code-style
- [X] My changes are relevant to the project

# Any other information (e.g how to test the changes)

None

> Note: This is my first PR so hopefully i didn't mess something up 

.

### CODE REVIEW

**Review by OpenSourceSimon — APPROVED** [review]

**Comment by liamb13** [comment]
Please take another look at this, I did a boiled down test and found it has the potential for some unexpected results:

```
import re

string = "imo: imogen does not understand multimolecular chemistry."
if re.search("imo", string):
    string = string.replace("imo", "in my opinion")
print(string)
```

The expected result of this is:
in my opinion: imogen does not understand multimolecular chemistry.

The actual result is:
in my opinion: in my opiniongen does not understand multin my opinionlecular chemistry.

File: TTS/engine_wrapper.py [hunk 1]
```diff
@@ -69,12 +70,35 @@ def add_periods(
...
 
     def run(self) -> Tuple[int, int]:
         Path(self.path).mkdir(parents=True, exist_ok=True)
         print_step("Saving Text to MP3 files...")
 
         self.add_periods()
+
+        if settings.config["settings"]["tts"]["replace_acronyms"]:
+            self.replace_acronyms()
+
```
Comment (electro199) [inline TTS/engine_wrapper.py:101]: don't do it like that dont place your code in wrapper instead do it process text functon

File: utils/acronyms.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    "__comment": "You can add more acronyms here as needed.",
+    "AITA": "Am I The Asshole",
+    "AITAH": "Am I The Asshole",
+    "NTA": "Not The Asshole",
+    "YTA": "Yes The Asshole",
+    "DEA": "Does Anyone Else",
+    "iirc": "If I Remember Correctly",
+    "imo": "In My Opinion",
+    "fyi": "For your information"
+}
```
Comment (electro199) [inline utils/acronyms.json:11]: dont put curse word

**Review by electro199 — COMMENTED** [review]
Need redesign

**Review by OpenSourceSimon — CHANGES_REQUESTED** [review]
See comments from others

**Comment by KingCaliph** [comment]
Did this ever get fully finished? I add the necessary code but ran into errors. I changed and added code where it seemed needed, and it did seem to fix certain initial errors, but I just kept running into more problems. Was this idea closed because people got bored with it, or is there some issue on my part?

**Comment by OpenSourceSimon** [comment]
It was closed because it has too much issues

**Pull request closed without merging**